Tests/test_image.py

```diff
@@ -138,21 +154,26 @@ def test_stringio(self):
             with Image.open(io.StringIO()):
                 pass
 
-    def test_pathlib(self, tmp_path):
-        from PIL.Image import Path
-
-        with Image.open(Path("Tests/images/multipage-mmap.tiff")) as im:
+    @pytest.mark.parametrize("PathCls", (
+        pathlib.Path,
+        MyStrPathLike,  # Returns `str` on `os.fspath`
+        MyBytesPathLike,  # Returns `bytes` on `os.fspath`
+        str,
+        os.fsencode,  # Converts path to `bytes`
+    ))
+    def test_path_like(self, tmp_path, PathCls):
+        with Image.open(PathCls("Tests/images/multipage-mmap.tiff")) as im:
```

Comment:
@radarhere: This test when using `MyBytesPathLike` or `os.fsencode` (ie `bytes`) as `PathCls` causes…